crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -0,0 +1,1006 @@
+        let obj2 = self.nodes.get_mut(idx).unwrap(); //TODO-RIA can we avoid this? and simply modify obj?
+        match obj2 {
+            node::NodeObj::Obj(o) => o.id = idx,
+            node::NodeObj::Instr(i) => {
+                i.idx = idx;
+                let cb = self.get_current_block_mut();
+                cb.instructions.push(idx);
+            }
+            node::NodeObj::Const(c) => c.id = idx,
+        }
+
+        idx
+    }
+
+    pub fn find_variable(&self, definition: &Option<IdentId>) -> Option<&node::Variable> {
+        if definition.is_none() {
+            return None;
+        }
+        for (_, o) in &self.nodes {
+            if let node::NodeObj::Obj(v) = o {
+                if v.def == *definition {
+                    return Some(v);
+                }
+            }
+        }
+        None
+    }
+
+    pub fn find_const(&self, value: &BigUint) -> Option<arena::Index> {
+        //TODO We should map constant values to id
+        for (idx, o) in &self.nodes {
+            if let node::NodeObj::Const(c) = o {
+                if c.value == *value {
+                    return Some(idx);
+                }
+            }
+        }
+        None
+    }
+
+    pub fn dummy_id() -> arena::Index {
+        arena::Index::from_raw_parts(std::usize::MAX, 0)
+    }
+
+    pub fn dummy(&self) -> arena::Index {
+        IRGenerator::dummy_id()
+    }
+
+    pub fn get_object(&self, idx: arena::Index) -> Option<&node::NodeObj> {
+        self.nodes.get(idx)
+    }
+
+    pub fn get_mut_object(&mut self, idx: arena::Index) -> Option<&mut node::NodeObj> {
+        self.nodes.get_mut(idx)
+    }
+
+    fn get_object_type(&self, idx: arena::Index) -> node::ObjectType {
+        self.get_object(idx).unwrap().get_type()
+    }
+
+    //Returns the object value if it is a constant, None if not. TODO: handle types
+    pub fn get_as_constant(&self, idx: arena::Index) -> Option<FieldElement> {
+        if let Some(node::NodeObj::Const(c)) = self.get_object(idx) {
+            return Some(FieldElement::from_be_bytes_reduce(&c.value.to_bytes_be()));
+        }
+        None
+    }
+
+    pub fn get_mut_instruction(&mut self, idx: arena::Index) -> &mut node::Instruction {
+        if let Some(node::NodeObj::Instr(i)) = self.get_mut_object(idx) {
+            return i;
+        }
+        unreachable!("Index not found or not an instruction");
+    }
+
+    pub fn get_as_instruction(&self, idx: arena::Index) -> Option<&node::Instruction> {
+        if let Some(node::NodeObj::Instr(i)) = self.get_object(idx) {
+            return Some(i);
+        }
+        None
+    }
+
+    pub fn get_as_mut_instruction(&mut self, idx: arena::Index) -> Option<&mut node::Instruction> {
+        if let Some(node::NodeObj::Instr(i)) = self.get_mut_object(idx) {
+            return Some(i);
+        }
+        None
+    }
+
+    //todo handle errors
+    fn get_instruction(&self, idx: arena::Index) -> &node::Instruction {
```

Comment:
This function can re-use the above instruction to avoid repeating code: `self.get_as_instruction(idx).expect("Index not found or not an instruction")`
Also, to follow rust's naming scheme with `get` more closely, perhaps `get_instruction` should return the Option, and this version should be `unwrap_instruction`? Or we could have `try_get_instruction` and `get_instruction`.

Reply:
Done, I used try_get_instruction and get_instruction.